uploaders/aabuploader.go

```diff
@@ -3,89 +3,54 @@ package uploaders
+func (u *Uploader) DeployAAB(item deployment.DeployableItem, artifacts []string, buildURL, token string) (ArtifactURLs, error) {
 	pth := item.Path
-	aabInfo, err := androidartifact.GetAABInfo(bt, pth)
+
+	aabInfo, err := u.androidParser.ParseAABData(pth)
 	if err != nil {
 		return ArtifactURLs{}, err
 	}
 
-	appInfo := map[string]interface{}{
-		"package_name":    aabInfo.PackageName,
-		"version_code":    aabInfo.VersionCode,
-		"version_name":    aabInfo.VersionName,
-		"app_name":        aabInfo.AppName,
-		"min_sdk_version": aabInfo.MinSDKVersion,
-	}
-
-	log.Printf("aab infos: %v", appInfo)
+	u.logger.Printf("aab infos: %v", aabInfo.AppInfo)
 
-	if aabInfo.PackageName == "" {
-		log.Warnf("Package name is undefined, AndroidManifest.xml package content:\n%s", aabInfo.RawPackageContent)
+	if aabInfo.AppInfo.PackageName == "" {
+		u.logger.Warnf("Package name is undefined, AndroidManifest.xml package content:\n%s", aabInfo.AppInfo.RawPackageContent)
 	}
 
-	if aabInfo.VersionCode == "" {
-		log.Warnf("Version code is undefined, AndroidManifest.xml package content:\n%s", aabInfo.RawPackageContent)
+	if aabInfo.AppInfo.VersionCode == "" {
+		u.logger.Warnf("Version code is undefined, AndroidManifest.xml package content:\n%s", aabInfo.AppInfo.RawPackageContent)
 	}
 
-	if aabInfo.VersionName == "" {
-		log.Warnf("Version name is undefined, AndroidManifest.xml package content:\n%s", aabInfo.RawPackageContent)
+	if aabInfo.AppInfo.VersionName == "" {
+		u.logger.Warnf("Version name is undefined, AndroidManifest.xml package content:\n%s", aabInfo.AppInfo.RawPackageContent)
 	}
 
-	if aabInfo.MinSDKVersion == "" {
-		log.Warnf("Min SDK version is undefined, AndroidManifest.xml package content:\n%s", aabInfo.RawPackageContent)
+	if aabInfo.AppInfo.MinSDKVersion == "" {
+		u.logger.Warnf("Min SDK version is undefined, AndroidManifest.xml package content:\n%s", aabInfo.AppInfo.RawPackageContent)
 	}
 
-	if aabInfo.AppName == "" {
-		log.Warnf("App name is undefined, AndroidManifest.xml package content:\n%s", aabInfo.RawPackageContent)
-	}
-
-	fileSize, err := fileSizeInBytes(pth)
-	if err != nil {
-		return ArtifactURLs{}, fmt.Errorf("failed to get apk size, error: %s", err)
-	}
-
-	info := androidartifact.ParseArtifactPath(pth)
-
-	aabInfoMap := map[string]interface{}{
-		"file_size_bytes": fmt.Sprintf("%d", fileSize),
-		"app_info":        appInfo,
-		"module":          info.Module,
-		"product_flavour": info.ProductFlavour,
-		"build_type":      info.BuildType,
-	}
-
-	signature, err := androidsignature.Read(pth)
-	if err != nil {
-		log.Warnf("Failed to read signature: %s", err)
+	if aabInfo.AppInfo.AppName == "" {
+		u.logger.Warnf("App name is undefined, AndroidManifest.xml package content:\n%s", aabInfo.AppInfo.RawPackageContent)
 	}
-	aabInfoMap["signed_by"] = signature
 
-	splitMeta, err := androidartifact.CreateSplitArtifactMeta(pth, artifacts)
+	splitMeta, err := androidartifact.CreateSplitArtifactMeta(u.logger, pth, artifacts)
```

Comment:
Why isn't the split meta creation part of the lib?

Reply:
The main artifact parser logic which is purely moved to the libs is about only individual artifacts. Split metadata uses multiple artifacts which is also a client-side decision. For now, I prefer to leave this to the deploy step, because on the Release Management side, we will not use this feature. We will probably support AAB over-the-air installation in our way.